### ISSUE TITLE
Flag duplicate patient profiles in patient self-registration UI

### ISSUE THREAD

**Issue by katiealoisi** [CDCgov/prime-simplereport#1379]
Spec: https://docs.google.com/document/d/1gVutot1vYrHHQYg6ExeTA6bhWib3YrE_77C1VItzAqk/edit  
Figma design: https://www.figma.com/file/NPSPM5SBzQ7Ra3mmr9M9cB/SimpleReport-Public-Site?node-id=3434%3A11296

Now that the patient self-registration is live we are seeing patients register themselves multiple times. To prevent duplicate profiles we will let users know they are already registered if there is an exact match on first name, last name, DOB, and zip code.

- [ ] In self-registration, if the profile is determined to be a duplicate the user will not be allowed to proceed with registering.

##Screenshots
Figma design: https://www.figma.com/file/NPSPM5SBzQ7Ra3mmr9M9cB/SimpleReport-Public-Site?node-id=3434%3A11296

See "Patient self-registration flow" section

**Comment by vz3** [comment]
Could we add this to the scope of this story:
- How can users add patients from the test queue page?

**Comment by aliciabeckett-gov** [comment]
We should rate limit this for sure!

**Comment by aliciabeckett-gov** [comment]
@vz3 that's a separate story. Let's keep them separate to keep the scope down.

**Comment by aliciabeckett-gov** [comment]
Blocked on Design

**Comment by kenieh** [comment]
Added the figma link. I'll have the spacing specs and state changes ready before our next grooming

**Comment by nickscialli-usds** [comment]
@aliciabeckett-gov Everything here needs Spanish translation!

**Comment by nickscialli-usds** [comment]
@kenieh and I paired on this a bit and have come to the conclusion that it's probably not worth breaking the form up into sections for now.

As designed, with the General and Contact information on the first page, the form validation becomes super tricky and the user only saves five radio button clicks if they are indeed a duplicate.

So it seems like the best option is to adopt the planned "Add Patient" flow of just checking if the user is a duplicate if any of the necessary fields (first name, last name, birthdate, zip) are blurred.
